Chap_API_Sync_Access.tex

```diff
@@ -1,171 +1,419 @@
+A client may call \refapi{PMIx_Put} as many times as necessary and the data is not available to other processes until explicitly committed.
+A client must call \refapi{PMIx_Commit} to make accessible all key-value pairs previously put by this process to all other processes in the \ac{PMIx} universe.
+This put and commit pattern provides implementors the opportunity to make individual \refapi{PMIx_Put} calls efficient local operations, and then make the whole set of key-value pairs accessible in a single step.
+
+\ac{PMIx} clients can access the key-value pairs associated with any process data realm 
+in the \ac{PMIx} universe (including the calling process) by passing the specific process 
+name of the target process to the \refapi{PMIx_Get} and \refapi{PMIx_Get_nb} functions.
+The \ac{PMIx} server local to the calling process will retrieve that key-value pair from 
+the \ac{PMIx} server associated with the target process.
+Clients can also access session, job, application, node, and namespace level information 
+by using the \refapi{PMIx_Get} and \refapi{PMIx_Get_nb} functions as shown in Section 
+\ref{chap:api_rsvd_keys:getex}.
+% and summarized in Table~\ref{tab:key-value-realms}.  (Add in next release?)
+The completion semantics for \refapi{PMIx_Get} and \refapi{PMIx_Get_nb} differ 
+depending on the type of key and its availability.  See Sections 
+\ref{chap:data_sharing:retrules} and \ref{chap:api_rsvd_keys:retrules}.
+For example, if a non-reserved key is not available, the \refapi{PMIx_Get} or \refapi{PMIx_Get_nb} call 
+will not complete, by default, until that key-value pair becomes available.
 
-Applications may need to synchronize their operations at various points in
-their execution. Depending on a variety of factors (e.g., the programming
-model and where the synchronization point lies), the application may choose to
-execute the operation using \ac{PMIx}. This is particularly useful in
-situations where communication by other means is not yet available since
-\ac{PMIx} relies on the host environment's infrastructure for such operations.
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\section{Non-reserved keys}
+\label{chap:data_sharing:non_rsvd_keys}
 
-Synchronization operations also offer an opportunity for processes to exchange
-data at a known point in their execution. Where required, this can include
-information on communication endpoints for subsequent wireup of various
-messaging protocols.
+\emph{Non-reserved keys} are keys whose string representation begin with a
+prefix other than \code{"pmix"}. Such keys are defined by a
+client when information needs to be exchanged between processes, for example, 
+where connection information is required and the host environment does not
+support the \refterm{instant on} option) or where the host environment does not
+provide a required piece of data. Other than the prefix, there are no
+restrictions on the use or content of non-reserved keys.
+
```

Comment:
Question: Can you have a key "foo" defined in both the LOCAL and REMOTE scopes possibly with different values?
Answer:
 * ~Believe that is not allowed. Need to confirm that this clarification is preserved - possibly in the scope section.~
 * Actually, it is allowed to have the key posted to both scopes.